src/internal/name.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * This file contains the underlying implementations for exposed API surface in
+ * the {@link api/name}. By moving the methods out into a separate file,
+ * other namespaces and processes can access these methods without depending on the entire
+ * name namespace and without having a dependency cycle error.
+ */
+
+import { AptosConfig } from "../api/aptos_config";
+import { U64 } from "../bcs";
+import { MoveOption, MoveString } from "../bcs/serializable/move-structs";
+import { Account, AccountAddress } from "../core";
+import { GenerateTransactionOptions, SingleSignerTransaction } from "../transactions/types";
+import { HexInput, MoveAddressType } from "../types";
+import { Network } from "../utils/apiEndpoints";
+import { generateTransaction } from "./transaction_submission";
+
+export const RouterAddressLookup: Record<Network, MoveAddressType> = {
+  testnet: "0x5f8fd2347449685cf41d4db97926ec3a096eaf381332be4f1318ad4d16a8497c",
+  mainnet: "0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c",
+  devnet: "" as MoveAddressType,
+  local: "" as MoveAddressType,
+  custom: "" as MoveAddressType,
+};
+
+function getRouterAddress(aptosConfig: AptosConfig): MoveAddressType {
+  const address = aptosConfig.ansRouter || RouterAddressLookup[aptosConfig.network];
+  if (!address) throw new Error("To use the Name API, you must provide a router address in the AptosConfig.");
+  return address;
+}
+
+function validateMultipleOfYearInSeconds(seconds: number): void {
+  if (seconds % 31536000 !== 0 || seconds === 0) {
+    throw new Error("Registration duration must be a multiple of 1 year in seconds, greater than 0");
+  }
+}
+
+export async function mintDomainName({
+  aptosConfig,
+  sender,
+  domainName,
+  registrationDuration,
+  targetAddress,
+  toAddress,
+  options,
+}: {
+  aptosConfig: AptosConfig;
+  sender: Account;
+  domainName: string;
+  registrationDuration: 31536000;
+  targetAddress?: HexInput;
+  toAddress?: HexInput;
+  options?: GenerateTransactionOptions;
+}): Promise<SingleSignerTransaction> {
+  const routerAddress = getRouterAddress(aptosConfig);
+  validateMultipleOfYearInSeconds(registrationDuration);
+
+  const transaction = await generateTransaction({
+    aptosConfig,
+    sender: sender.accountAddress.toString(),
+    data: {
+      function: `${routerAddress}::router::register_domain`,
+      typeArguments: [],
+      arguments: [
+        new MoveString(domainName),
+        new U64(registrationDuration),
+        new MoveOption(targetAddress ? AccountAddress.fromHexInput(targetAddress) : null),
+        new MoveOption(toAddress ? AccountAddress.fromHexInput(toAddress) : null),
```

Comment:
Just double checking I was correctly using `MoveOption` here. The contract doesn't require these fields.